File: .claude/e2e-smoke.mjs
import { chromium } from '@playwright/test'
import { mkdirSync, writeFileSync } from 'node:fs'
import path from 'node:path'

const BASE = 'http://localhost:3000'
const OUT = path.resolve('.claude/screenshots')
mkdirSync(OUT, { recursive: true })

const ROUTES = [
  ['dashboard', '/'],
  ['mis-tareas', '/mis-tareas'],
  ['tareas', '/tareas'],
  ['colaborativas', '/colaborativas'],
  ['nueva-tarea', '/nueva-tarea'],
  ['calendario', '/calendario'],
  ['notfound', '/esta-ruta-no-existe'],
]
const VIEWPORTS = [['desktop', 1440, 900], ['mobile', 390, 844]]

const report = { errors: [], routes: {}, interactions: {} }

const browser = await chromium.launch({ headless: true })

for (const [vname, width, height] of VIEWPORTS) {
  const ctx = await browser.newContext({ viewport: { width, height } })
  const page = await ctx.newPage()
  let msgs = []
  page.on('console', (m) => msgs.push({ type: m.type(), text: m.text() }))
  page.on('pageerror', (e) => report.errors.push({ viewport: vname, error: String(e) }))

  for (const [name, p] of ROUTES) {
    try {
      await page.goto(BASE + p, { waitUntil: 'domcontentloaded', timeout: 30000 })
      await page.waitForLoadState('networkidle', { timeout: 15000 }).catch(() => {})
    } catch (e) {
      report.errors.push({ viewport: vname, route: p, nav_error: String(e) })
      continue
    }
    await page.waitForTimeout(800)
    const shot = path.join(OUT, `${vname}-${name}.png`)
    try { await page.screenshot({ path: shot, fullPage: true }) } catch {}
    const errs = msgs.filter((m) => m.type === 'error')
    const warns = msgs.filter((m) => m.type === 'warning')
    report.routes[p] = report.routes[p] || {}
    report.routes[p][vname] = {
      console_errors: errs.slice(-10),
      console_warnings: warns.slice(-5),
      title: await page.title(),
      url: page.url(),
    }
    msgs = []
  }
  await ctx.close()
}

// Interactions on desktop
const ctx = await browser.newContext({ viewport: { width: 1440, height: 900 } })
const page = await ctx.newPage()
const rec = (n, ok, d = '') => (report.interactions[n] = { ok, detail: d })

try {
  await page.goto(BASE, { waitUntil: 'networkidle', timeout: 20000 })
  await page.waitForTimeout(600)
  const toggle = page.locator('[aria-label*="tema" i], [aria-label*="theme" i]').first()
  if (await toggle.count() > 0) {
    await toggle.click()
    await page.waitForTimeout(400)
    await page.screenshot({ path: path.join(OUT, 'int-theme-dark.png') })
    rec('theme-toggle', true, 'clicked')
  } else rec('theme-toggle', false, 'selector not found')
} catch (e) { rec('theme-toggle', false, String(e)) }

try {
  await page.goto(BASE, { waitUntil: 'networkidle', timeout: 20000 })
  await page.waitForTimeout(500)
  const bell = page.locator('[aria-label*="notifica" i]').first()
  if (await bell.count() > 0) {
    await bell.click()
    await page.waitForTimeout(500)
    await page.screenshot({ path: path.join(OUT, 'int-notifications.png') })
    rec('notification-bell', true)
  } else rec('notification-bell', false, 'not found')
} catch (e) { rec('notification-bell', false, String(e)) }

try {
  await page.goto(BASE + '/calendario', { waitUntil: 'networkidle', timeout: 20000 })
  await page.waitForTimeout(800)
  await page.screenshot({ path: path.join(OUT, 'int-calendar.png') })
  const day = page.locator('button:has-text("15"), [role="button"]:has-text("10")').first()
  if (await day.count() > 0) {
    await day.click()
    await page.waitForTimeout(500)
    await page.screenshot({ path: path.join(OUT, 'int-calendar-day.png') })
    rec('calendar-day-click', true)
  } else rec('calendar-day-click', false, 'no day button')
} catch (e) { rec('calendar-day-click', false, String(e)) }

try {
  await page.goto(BASE + '/tareas', { waitUntil: 'networkidle', timeout: 20000 })
  await page.waitForTimeout(600)
  const row = page.locator('tbody tr, [role="row"]').nth(1)
  if (await row.count() > 0) {
    await row.click()
    await page.waitForTimeout(700)
    await page.screenshot({ path: path.join(OUT, 'int-task-detail.png') })
    rec('open-task-detail', true)
  } else rec('open-task-detail', false, 'no row')
} catch (e) { rec('open-task-detail', false, String(e)) }

try {
  await page.goto(BASE + '/nueva-tarea', { waitUntil: 'networkidle', timeout: 20000 })
  await page.waitForTimeout(700)
  const fields = await page.locator('input, textarea, [role="combobox"]').count()
  await page.screenshot({ path: path.join(OUT, 'int-new-task.png') })
  rec('new-task-form', fields > 2, `${fields} fields`)
} catch (e) { rec('new-task-form', false, String(e)) }

try {
  await page.goto(BASE, { waitUntil: 'networkidle', timeout: 20000 })
  await page.waitForTimeout(500)
  await page.keyboard.press('Control+b')
  await page.waitForTimeout(400)
  await page.screenshot({ path: path.join(OUT, 'int-sidebar-collapsed.png') })
  rec('sidebar-collapse', true)
} catch (e) { rec('sidebar-collapse', false, String(e)) }

try {
  await page.goto(BASE, { waitUntil: 'networkidle', timeout: 20000 })
  await page.waitForTimeout(500)
  await page.keyboard.press('Control+k')
  await page.waitForTimeout(500)
  const palette = page.locator('[role="dialog"], [cmdk-root]').first()
  rec('command-palette', await palette.count() > 0)
  if (await palette.count() > 0) {
    await page.screenshot({ path: path.join(OUT, 'int-command-palette.png') })
  }
} catch (e) { rec('command-palette', false, String(e)) }

await ctx.close()
await browser.close()

writeFileSync('.claude/e2e-report.json', JSON.stringify(report, null, 2))

const routesWithErrors = Object.entries(report.routes)
  .filter(([, v]) => Object.values(v).some((vv) => vv.console_errors.length))
  .map(([k]) => k)

console.log(JSON.stringify({
  page_errors: report.errors.length,
  page_error_samples: report.errors.slice(0, 5),
  routes_with_console_errors: routesWithErrors,
  interactions: report.interactions,
}, null, 2))

File: .claude/e2e-smoke.py
"""Smoke test for OTC maqueta. Visits every route, captures console errors + screenshots."""
from playwright.sync_api import sync_playwright
import sys, os, json

BASE = "http://localhost:3000"
OUT = r"C:\Users\mathi\maqueta-OTC\.claude\screenshots"
os.makedirs(OUT, exist_ok=True)

ROUTES = [
    ("dashboard", "/"),
    ("mis-tareas", "/mis-tareas"),
    ("tareas", "/tareas"),
    ("colaborativas", "/colaborativas"),
    ("nueva-tarea", "/nueva-tarea"),
    ("calendario", "/calendario"),
    ("notfound", "/esta-ruta-no-existe"),
]

VIEWPORTS = [
    ("desktop", 1440, 900),
    ("mobile", 390, 844),
]

report = {"errors": [], "warnings": [], "routes": {}}

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    for vname, w, h in VIEWPORTS:
        ctx = browser.new_context(viewport={"width": w, "height": h})
        page = ctx.new_page()
        console_msgs = []
        page.on("console", lambda m: console_msgs.append({"type": m.type, "text": m.text}))
        page.on("pageerror", lambda e: report["errors"].append({"viewport": vname, "error": str(e)}))

        for name, path in ROUTES:
            try:
                page.goto(BASE + path, wait_until="domcontentloaded", timeout=30000)
                page.wait_for_load_state("networkidle", timeout=15000)
            except Exception as e:
                report["errors"].append({"viewport": vname, "route": path, "nav_error": str(e)})
                continue
            page.wait_for_timeout(800)
            shot = os.path.join(OUT, f"{vname}-{name}.png")
            try:
                page.screenshot(path=shot, full_page=True)
            except Exception as e:
                report["errors"].append({"viewport": vname, "route": path, "screenshot_error": str(e)})
            # capture per-route console
            errs = [m for m in console_msgs if m["type"] == "error"]
            warns = [m for m in console_msgs if m["type"] == "warning"]
            report["routes"].setdefault(path, {})[vname] = {
                "console_errors": errs[-10:],
                "console_warnings": warns[-5:],
                "title": page.title(),
                "url": page.url,
            }
            console_msgs.clear()
        ctx.close()

    # --- Interaction tests on desktop ---
    ctx = browser.new_context(viewport={"width": 1440, "height": 900})
    page = ctx.new_page()
    interactions = {}

    def record(name, ok, detail=""):
        interactions[name] = {"ok": ok, "detail": detail}

    # 1. Dashboard loads, toggle theme
    try:
        page.goto(BASE, wait_until="networkidle", timeout=20000)
        page.wait_for_timeout(600)
        # theme toggle — try aria-label
        toggle = page.locator('[aria-label*="tema" i], [aria-label*="theme" i], button:has-text("Tema")').first
        if toggle.count() > 0:
            toggle.click()
            page.wait_for_timeout(400)
            page.screenshot(path=os.path.join(OUT, "int-theme-dark.png"))
            record("theme-toggle", True, "clicked")
        else:
            record("theme-toggle", False, "selector not found")
    except Exception as e:
        record("theme-toggle", False, str(e))

    # 2. Notification bell
    try:
        page.goto(BASE, wait_until="networkidle", timeout=20000)
        page.wait_for_timeout(500)
        bell = page.locator('[aria-label*="notifica" i], button:has([class*="Bell"])').first
        if bell.count() > 0:
            bell.click()
            page.wait_for_timeout(500)
            page.screenshot(path=os.path.join(OUT, "int-notifications.png"))
            record("notification-bell", True)
        else:
            record("notification-bell", False, "not found")
    except Exception as e:
        record("notification-bell", False, str(e))

    # 3. Calendar
    try:
        page.goto(BASE + "/calendario", wait_until="networkidle", timeout=20000)
        page.wait_for_timeout(800)
        page.screenshot(path=os.path.join(OUT, "int-calendar.png"))
        # try clicking a day with tasks
        day = page.locator('[role="button"]:has-text("1"), button:has-text("15")').first
        if day.count() > 0:
            day.click()
            page.wait_for_timeout(500)
            page.screenshot(path=os.path.join(OUT, "int-calendar-day.png"))
            record("calendar-day-click", True)
        else:
            record("calendar-day-click", False, "no day button found")
    except Exception as e:
        record("calendar-day-click", False, str(e))

    # 4. Tasks list → open detail
    try:
        page.goto(BASE + "/tareas", wait_until="networkidle", timeout=20000)
        page.wait_for_timeout(600)
        # click first row
        row = page.locator('button, [role="row"], tr').nth(1)
        if row.count() > 0:
            row.click()
            page.wait_for_timeout(600)
            page.screenshot(path=os.path.join(OUT, "int-task-detail.png"))
            record("open-task-detail", True)
        else:
            record("open-task-detail", False, "no row")
    except Exception as e:
        record("open-task-detail", False, str(e))

    # 5. Nueva tarea form visible
    try:
        page.goto(BASE + "/nueva-tarea", wait_until="networkidle", timeout=20000)
        page.wait_for_timeout(700)
        inputs = page.locator("input, textarea, select").count()
        page.screenshot(path=os.path.join(OUT, "int-new-task.png"))
        record("new-task-form", inputs > 2, f"{inputs} fields")
    except Exception as e:
        record("new-task-form", False, str(e))

    # 6. Sidebar collapse via shortcut
    try:
        page.goto(BASE, wait_until="networkidle", timeout=20000)
        page.wait_for_timeout(500)
        page.keyboard.press("Control+b")
        page.wait_for_timeout(400)
        page.screenshot(path=os.path.join(OUT, "int-sidebar-collapsed.png"))
        record("sidebar-collapse", True)
    except Exception as e:
        record("sidebar-collapse", False, str(e))

    report["interactions"] = interactions
    ctx.close()
    browser.close()

with open(r"C:\Users\mathi\maqueta-OTC\.claude\e2e-report.json", "w", encoding="utf-8") as f:
    json.dump(report, f, indent=2, ensure_ascii=False)

print(json.dumps({
    "page_errors": len(report["errors"]),
    "routes_with_console_errors": [
        r for r, v in report["routes"].items()
        if any(vv["console_errors"] for vv in v.values())
    ],
    "interactions": interactions,
}, indent=2, ensure_ascii=False))
